packages/tailwindcss/src/index.ts

```diff
@@ -501,6 +541,30 @@ async function parseCss(
     node.context = {}
   }
 
+  // Replace the `@variant` at-rules with the actual variant rules.
+  if (variantNodes.length > 0) {
+    for (let variantNode of variantNodes) {
+      // Starting with the `&` rule node
+      let node = styleRule('&', variantNode.nodes)
+
+      let variant = variantNode.params
+
+      let variantAst = designSystem.parseVariant(variant)
+      if (variantAst === null) {
+        throw new Error(`Cannot use \`@variant\` with unknown variant: ${variant}`)
+      }
```

Comment:
Could someone theoretically register a `hover:focus` variant through configs and then try using that with `@variant`?

We might want explicitly to error on uses of `:` if that's possible

Reply:
So you can technically add a custom variant with a `:`, but you will never be able to use it because we split it on `:` anyway.

Added some validation (relatively strict) which we can loosen up down the line if needed.